kubernetes/destinationpolicy_type.go

```diff
@@ -0,0 +1,120 @@
+package kubernetes
+
+import (
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// DestinationPolicy is one of the specific types defined by Istio as a Kubernetes extension.
+// Preliminar rule is to define one file per type.
+// types.go will collect common/shared types.
+// This type is extracted from Istio Pilot as models used are not public and it doesn't make sense to fetch all
+// Istio project as a dependency.
+// Reference: https://github.com/istio/istio/blob/master/pilot/pkg/config/kube/crd/types.go
+
+const DestinationPolicies = "destinationpolicies"
+const DestinationPolicyType = "DestinationPolicy"
+const DestinationPolicyLabel = "destination-policy"
+
+// DestinationPolicy is the generic Kubernetes API object wrapper
+type DestinationPolicy struct {
+	meta_v1.TypeMeta   `json:",inline"`
+	meta_v1.ObjectMeta `json:"metadata"`
+	Spec               map[string]interface{} `json:"spec"`
+}
+
+// DestinationPolicyList is the generic Kubernetes API list wrapper
+type DestinationPolicyList struct {
+	meta_v1.TypeMeta `json:",inline"`
+	meta_v1.ListMeta `json:"metadata"`
+	Items            []DestinationPolicy `json:"items"`
+}
+
+// GetSpec from a wrapper
+func (in *DestinationPolicy) GetSpec() map[string]interface{} {
```

Comment:
I see that these Getters/Setters are not used here. Are they going to be used for sure in a really close future? IMO, if they are not, I would remove them.

Reply:
routerule_type.go and destinationpolicy_type.go are extracted/cloned from Istio code.
Some of the methods are used in our code, and probably others as well when these types are registered.
This suggestion to refactor is good, but I would say to wait until we consolidate better the code.

Reply:
good to know. I recall now you already told that before. Let's see where codebase goes and then decide.